src/main/kotlin/io/openfuture/state/repository/OffsetRepository.kt

```diff
@@ -0,0 +1,20 @@
+package io.openfuture.state.repository
+
+import io.openfuture.state.model.BlockchainType
+import org.springframework.data.redis.core.ReactiveRedisTemplate
+import org.springframework.data.redis.core.ReactiveValueOperations
+import org.springframework.stereotype.Repository
+import reactor.core.publisher.Mono
+
+@Repository
+class OffsetRepository(private val offsetRedisTemplate: ReactiveRedisTemplate<BlockchainType, Long>) {
+    private val valueOperations: ReactiveValueOperations<BlockchainType, Long> = offsetRedisTemplate.opsForValue()
+
+    fun getCurrent(blockchain: BlockchainType, defaultValueIfEmpty: Long = 0): Mono<Long> {
```

Comment:
you should not pass defaultValueIfEmpty as a parameter. This logic should be encapsulated.